homework/index.js

```diff
@@ -30,17 +26,91 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+  function renderError(error) {
+    const root = document.getElementById('root');
+    root.innerHTML = '';
+    createAndAppend('h1', root, { text: error.message, class: `alert-error` });
+  }
+
+  function addRow(tbody, label, value) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${label}:`, class: 'label' });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepos(repo, div) {
+    div.innerHTML = '';
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Name', '');
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: `_blank`,
     });
+    addRow(tbody, 'Description', repo.description);
+    addRow(tbody, 'Forks', repo.forks);
+    addRow(tbody, 'Updated', new Date(repo.updated_at).toLocaleString('en-GB'));
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  async function renderContributions(repo, ul) {
+    ul.innerHTML = '';
+    createAndAppend('li', ul, { text: `Contributors:` });
+
+    try {
+      const contributors = await fetchJSON(repo.contributors_url);
+      contributors.forEach(contributor => {
+        const li = createAndAppend('li', ul);
+        const a = createAndAppend('a', li, { href: contributor.html_url, target: '_blank' });
+        const table = createAndAppend('table', a);
+        const tbody = createAndAppend('tbody', table);
+        const tr1 = createAndAppend('tr', tbody);
+        createAndAppend('img', tr1, { src: contributor.avatar_url });
```

Comment:
This created invalid HTML. A `tr` element can only have `td` and `th` children. 

```html
<ul id="list-contributions">
  <li>Contributors:</li>
  <li>
    <a href="https://github.com/gijscor" target="_blank">
      <table>
        <tbody>
          <tr>
            <img src="https://avatars0.githubusercontent.com/u/15912395?v=4" />  <-- invalid HTML
            <td>gijscor</td>
            <td>47</td>
          </tr>
        </tbody>
      </table></a
    >
  </li>
</ul>
```

Reply:
Thanks I fixed it